File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    text = String.trim(input)
    yelling? = text == String.upcase(text) and text != String.downcase(text)
    q? = String.ends_with?(text, "?")

    case {text, yelling?, q?} do
      {"", _, _} -> "Fine. Be that way!"
      {_, true, true} -> "Calm down, I know what I'm doing!"
      {_, false, true} -> "Sure."
      {_, true, _} -> "Whoa, chill out!"
      _ -> "Whatever."
    end
  end
end

File: python/bob/bob.py
def hey(phrase):
  text = phrase.strip()
  is_yelling = text == text.upper() and text != text.lower()

  if text == '':
    return 'Fine. Be that way!'
  elif text[-1] == '?':
    if is_yelling:
      return 'Calm down, I know what I\'m doing!'
    else:
      return 'Sure.'
  elif is_yelling:
    return 'Whoa, chill out!'
  else:
    return 'Whatever.'
